lib/tests/modules.sh

```diff
@@ -323,6 +323,7 @@ checkConfigError 'The option .* is used but not defined' config.foo ./freeform-l
 checkConfigOutput '^"24"$' config.foo ./freeform-lazyAttrsOf.nix ./freeform-unstr-dep-str.nix ./define-value-string.nix
 # submodules in freeformTypes should have their locations annotated
 checkConfigOutput '/freeform-submodules.nix"$' config.fooDeclarations.0 ./freeform-submodules.nix
+checkConfigOutput '/freeform-submodules.nix"$' config.fooDeclarationsLoc.0.file ./freeform-submodules.nix
```

Comment:
And this should be testable by using `options.declarationsWithLocations` instead of `config.`

Reply:
I wrote a bunch of tests that don't do this, but the funny submodules things cannot be avoided since they need to call a function; I do want to check they work because idk why not.